### DISCUSSION TITLE
Reading data failed! Inverter offline.

### DISCUSSION THREAD

**Discussion by VilemP** [wills106/homeassistant-solax-modbus#123] [Q&A]
Hi. 
I have G4 X3 Hybrid inverter connected via waveshare adapter.  Upon installation it works quite well. But after a couple of hours I get the error "Reading data failed! Inverter offline.". 
The Wveshare's leds are off.
I have to unplug the waveshare adapter for some time (more than just 5 minutes) and then plug it again for it to start working again.

I have already change the power source for the adapter, but that did not help. 
Any ideas what might be causing this? I can't really find anything.

Thanks. Great work, Wills106!

**Comment by infradom** [comment]
If all Waveshare leds are off, there must be something wrong with the waveshare or its power wires. If I disconnect my inverter from the Waveshare, some leds remain on. Are you using the rail-mounted version of the Waveshare?

**Reply by VilemP** [reply]
Hi, thanks.
Yes, This one: https://www.waveshare.com/wiki/RS485_TO_ETH_(B)#Documentation
It is really weird. I wanted to exclude other options before returning it back to the seller. 
I initially thought it was the power issue and so I changed the adapter I used to power it with, but the behavior remained. It runs for couple of hours, then with no apparent reason, I start getting those error messages in HA and when I go to check it, the leds are off.
What is even weirder for me is that it takes well over 5 minutes after unplugging the waveshare from the electricity for it to "turn on" again.

**Reply by infradom** [reply]
seems like temperature related, although mine does not get warm at all. Which DC voltage do you feed it ?

**Reply by VilemP** [reply]
checked the temperature and it seems to have more or less ambient temperature
I am using 12V to feed it. Should work with up to 24V so I guess this should not be an issue?

**Comment by VilemP** [comment]
Update: today it went offline for about 30 minutes and then - without any intervention - it came back online again :( Really don't know what to do, it is quite annoying as the Energy dashboard displays a crap then :-( Do you think it could be a faulty waveshare?

**Reply by wills106** [reply]
How quickly are you polling the Inverter?
I am just wondering if your are overloading it by polling every second for example?

**Reply by VilemP** [reply]
Thanks. That was my thought to. I started at 5 seconds. I have already changed that to 15s (via 10s), to no avail unfortunately. It died an hour ago again. Now waiting whether it will recover again by itself.

**Reply by wills106** [reply]
Can you still connect to the Waveshare Web UI. Or does the Waveshare fully crash?

**Reply by VilemP** [reply]
Nope. When it's crashed I can't. It has come back online after 80 minutes now without any intervention.

**Reply by wills106** [reply]
I don't own a Waveshare. It does point to it being faulty though if you can't even use the Web UI.

**Reply by infradom** [reply]
It is likely a bad unit, but it can also be a simple mechanical problem around the power connector (or an internally broken power lead). Simple things like temperature rises can break or make the connection.

**Reply by VilemP** [reply]
well, well, well.. I let it "run" for some time without tinkering with it
and I found out there is a pattern - it runs for 250 mins and then it does not for 85 mins and it again and again....
![image](https://user-images.githubusercontent.com/3738133/193267705-8bf41baa-4f63-46c1-9694-2afb17e371f6.png)

Weird.

**Reply by infradom** [reply]
Indeed, very weird. My suggestion that it is temperature related may not hold.
To be absolutely sure, you could use a hairdryer to try to influence the timing of the dropouts

**Reply by VilemP** [reply]
Anyways, thanks for trying to help. Much appreciated. I ordered another unit and we'll see whether it is Waveshare or not. Will keep you posted about the results.

**Reply by VilemP** [reply]
Haha. Well, that did not turn out well for me :-)

Just by accident I now found out that the plug that I use to power the waveshare from is managed by my waste water cleaner control panel and basically turned off regularly when the air pump which is normally plugged into is not needed 🤦‍♂️ So my problem actually was all the time that there was no power for waveshare at all.
Lucky me that I wanted to measure power consumption of another device from the same plug, thus discovering it's actually delivering no power. Arrrghhh....

Sorry for wasting your time guys. And thanks for trying to help.
Seems I will have a spare waveshare soon :-)
